frontend/src/components/Tags.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable react/jsx-props-no-spreading, @typescript-eslint/no-explicit-any */
+
+import { CloseIcon } from "@chakra-ui/icons";
+import {
+  Container,
+  FormControl,
+  FormLabel,
+  IconButton,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { components } from "react-select";
+import Creatable from "react-select/creatable";
+
+import tagAPIClient from "../APIClients/TagAPIClient";
+import { TagResponse } from "../types/TagTypes";
+import ConfirmationModal from "./common/ConfirmationModal";
+
+const EmptyTag = { value: "", label: "" };
+
+const customStyles = {
+  option: (provided: any) => ({
+    ...provided,
+    display: "flex",
+    paddingLeft: "5%",
+    justifyContent: "space-between",
+    ":hover": {
+      color: "gray.100",
+    },
+  }),
+  multiValueRemove: (provided: any, data: any) => ({
+    ...provided,
+    ":hover": {
+      backgroundColor: data.color,
+      color: "white",
+    },
+  }),
+};
+
+const Tags = (): React.ReactElement => {
+  const [tagOptions, setTagOptions] = useState<TagResponse[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [tagToDelete, setTagToDelete] = useState<TagResponse>(EmptyTag);
+
+  const onClose = () => setShowModal(false);
+
+  useEffect(() => {
+    tagAPIClient.getTags().then((newTags) => {
+      setTagOptions(newTags);
+      setIsLoading(false);
+    });
+  }, []);
+
+  const handleCreate = () => {
+    setIsLoading(true);
+
+    // TODO: Create and set tag
```

Comment:
Did you forget to implement this? The way create should work is it should be strictly frontend only so you can add a new tag to the array: TagOptions